src/format.cc

```diff
@@ -22,7 +24,17 @@ Format::Format(const Format&) = default;
 
 Format::~Format() = default;
 
-uint32_t Format::SizeInBytes() const {
+std::unique_ptr<Format> Format::Clone() const {
+  auto ret = MakeUnique<Format>();
```

Comment:
why not use the copy constructor?

`*ret = *this;`

Or even better:
`return MakeUnique<Format>(*this); ` 

Why do we need a clone method at all?